pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotSegmentUploadRestletResource.java

```diff
@@ -428,8 +428,20 @@ private Representation uploadSegment(File indexDir, File dataFile, String downlo
       throws ConfigurationException, IOException, JSONException {
     final SegmentMetadata metadata = new SegmentMetadataImpl(indexDir);
     final File tableDir = new File(baseDataDir, metadata.getTableName());
+    String tableName = metadata.getTableName();
     File segmentFile = new File(tableDir, dataFile.getName());
-    StorageQuotaChecker.QuotaCheckerResponse quotaResponse = checkStorageQuota(indexDir, metadata);
+    OfflineTableConfig offlineTableConfig = (OfflineTableConfig) ZKMetadataProvider
```

Comment:
We don't for checking quota. You can see that checkStorageQuota method signature is changed to take table config as a parameter.

That said, after checking, we will hit zk again during pinotHelixResourceManager.addSegment call below if the segment exists for refresh use case. Changing that is a bigger surgery since this is widely used api method